examples/inclined_plane.py

```diff
@@ -0,0 +1,130 @@
+print("generating simulated data using the true coefficient of friction %.3f" % mu0)
+N_obs = 20
+torch.manual_seed(2)
+observed_data = torch.cat([simulate(Variable(torch.Tensor([mu0])), noise_sigma=time_measurement_sigma)
+                           for _ in range(N_obs)])
+observed_mean = np.mean([T.data[0] for T in observed_data])
+
+
+# define model with uniform prior on mu and gaussian noise on the descent time
+def model(observed_data):
+    mu_prior = Uniform(Variable(torch.zeros(1)), Variable(torch.ones(1)))
+    mu = pyro.sample("mu", mu_prior)
+
+    def observe_T(T_obs, obs_name):
+        T_simulated = simulate(mu)
+        T_obs_dist = Normal(T_simulated, Variable(torch.Tensor([time_measurement_sigma])))
+        pyro.observe(obs_name, T_obs_dist, T_obs)
+
+    for i, T_obs in enumerate(observed_data):
+        observe_T(T_obs, "obs_%d" % i)
+
+    return mu
+
+
+def main():
+    parser = argparse.ArgumentParser(description="parse args")
+    parser.add_argument('-n', '--num-samples', default=500, type=int)
+    args = parser.parse_args()
+
+    # create an importance sampler (the prior is used as the proposal distribution)
+    posterior = Importance(model, guide=None, num_samples=args.num_samples)
+    # create a marginal object that consumes the raw execution traces provided by the importance sampler
+    marginal = Marginal(posterior)
+    # get posterior samples of mu (which is the return value of model)
+    print("doing importance sampling...")
+    posterior_samples = [marginal(observed_data) for i in range(args.num_samples)]
+
+    # calculate statistics over posterior samples
+    posterior_mean = torch.mean(torch.cat(posterior_samples))
+    posterior_std_dev = torch.std(torch.cat(posterior_samples), 0)
+
+    # report results
+    inferred_mu = posterior_mean.data[0]
+    inferred_mu_uncertainty = posterior_std_dev.data[0]
+    print("the coefficient of friction inferred by pyro is %.3f +- %.3f" %
+          (inferred_mu, inferred_mu_uncertainty))
+
+    # note that, given the finite step size in the simulator, the simulated descent times will
+    # not precisely match the numbers from the analytic result.
+    # in particular the first two numbers reported below should match each other pretty closely
+    # but will be systematically off from the third number
+    print("the mean observed descent time in the dataset is: %.4f seconds" % observed_mean)
+    print("the (forward) simulated descent time for the inferred (mean) mu is: %.4f seconds" %
+          simulate(posterior_mean).data[0])
+    print(("disregarding measurement noise, elementary calculus gives the descent time\n" +
+           "for the inferred (mean) mu as: %.4f seconds") % analytic_T(posterior_mean.data[0]))
+
+    """
+    ################## EXERCISE ###################
+    # vectorize the computations in this example! #
+    ###############################################
+    """
+
+
+if __name__ == '__main__':
```

Comment:
Consider moving arg parsing here and passing `args` to `main(args)`. That would help focus attention on the math in `main()`.

Reply:
thanks, done